File: datvar.rb
# Integer
my_int = 42
puts "Integer: #{my_int}"

# Float
my_float = 3.14
puts "Float: #{my_float}"

# Double (Ruby uses float for both float and double)
my_double = 3.14159
puts "Double: #{my_double}"

# String
my_string = "Hello, Ruby!"
puts "String: #{my_string}"

# Boolean
my_boolean = true
puts "Boolean: #{my_boolean}"

# Char (represented as a string of length 1 in Ruby)
my_char = 'A'
puts "Char: #{my_char}"

# Array
my_array = [1, 2, 3, 4, 5]
puts "Array: #{my_array.join(', ')}"

# Hash (Dictionary)
my_hash = { 'one' => 1, 'two' => 2, 'three' => 3 }
my_hash.each do |key, value|
  puts "Hash: #{key} => #{value}"
end

# Function
my_function = ->(x) { x * 2 }
puts "Function (2 * 3): #{my_function.call(3)}"

# Struct
Person = Struct.new(:name, :age)
my_person = Person.new('Alice', 25)
puts "Struct: Name = #{my_person.name}, Age = #{my_person.age}"

# Nil
my_nil = nil
puts "Nil: #{my_nil.inspect}"

File: datvar.clj
(ns data-types.core)

(defn -main []
  ;; Integer
  (def my-int 42)
  (println "Integer:" my-int)

  ;; Float
  (def my-float 3.14)
  (println "Float:" my-float)

  ;; Double
  (def my-double 3.14159)
  (println "Double:" my-double)

  ;; String
  (def my-string "Hello, Clojure!")
  (println "String:" my-string)

  ;; Boolean
  (def my-boolean true)
  (println "Boolean:" my-boolean)

  ;; Character
  (def my-char \A)
  (println "Char:" my-char)

  ;; Vector (Array equivalent)
  (def my-vector [1 2 3 4 5])
  (println "Vector:" my-vector)

  ;; List
  (def my-list '(1 2 3 4 5))
  (println "List:" my-list)

  ;; Set
  (def my-set #{1 2 3 4 5})
  (println "Set:" my-set)

  ;; Map (Dictionary)
  (def my-map {"one" 1 "two" 2 "three" 3})
  (println "Map:" my-map)

  ;; Function
  (def my-function (fn [x] (* x 2)))
  (println "Function (2 * 3):" (my-function 3))

  ;; Record (Struct equivalent)
  (defrecord Person [name age])
  (def my-person (->Person "Alice" 25))
  (println "Record: Name =" (:name my-person) "Age =" (:age my-person))

  ;; Nil
  (def my-nil nil)
  (println "Nil:" my-nil))

;; Run the main function
(-main)
